pyracf/common/security_admin.py

```diff
@@ -170,24 +172,49 @@ def __raw_dump(self) -> None:
         if self.__debug:
             # Note, since the hex dump is logged to the console,
             # secrets will be redacted.
-            self._logger.log_hex_dump(raw_result_xml, self.__secret_traits)
+            self._logger.log_hex_dump(
+                raw_result_xml,
+                self.__secret_traits,
+                self._racf_trait_and_message_key_map,
+            )
 
     # ============================================================================
     # Secrets Redaction
     # ============================================================================
     def __add_additional_secret_traits(self, additional_secret_traits: list) -> None:
         """Add additional fields to be redacted in logger output."""
+        unsupported_profile_types = ["permission", "groupConnection", "systemSettings"]
+        error_message = (
+            f"Cannot add specified additional secrets to '{self._profile_type}' object."
```

Comment:
You should be able to use `inspect.stack()` here to get the name of the "admin" class (i.e., `UserAdmin`): https://github.com/ambitus/pyracf/blob/dd23a06a2a886d7d9b122be36dfdade155917c89/pyracf/common/utilities/logger.py#L116

Also change the word `object` to `type` and change `Cannot` to `Unable`.

You should get an exception message like this when you are done:

```console
Unable to add specified additional secrets to 'UserAdmin' type.
```

Make similar corresponding changes to the unsupported profile type exception message.